app/controllers/course/material/folders_controller.rb

```diff
@@ -0,0 +1,52 @@
+class Course::Material::FoldersController < Course::ComponentController
+  load_and_authorize_resource :material_folder, through: :course, parent: false,
```

Comment:
It depends on when we want a folder that belongs to the course, or specifically material_folders. Since folders can be owned by other things (eg assessment)